src/ArrayReader.cpp

```diff
@@ -173,7 +181,15 @@ template<>
 void AppendArray<arrow::Type::INT32>(shared_ptr<arrow::Array> array_data, K k_array, size_t& index, TypeMappingOverride& type_overrides)
 {
   auto int32_array = static_pointer_cast<arrow::Int32Array>(array_data);
-  memcpy(kI(k_array), int32_array->raw_values(), int32_array->length() * sizeof(arrow::Int32Array::value_type));
+  if( type_overrides.null_mapping.have_int32 ){

```

Comment:
```suggestion
  if( type_overrides.null_mapping.have_int32 && int32_array->null_count() != 0 ){

```

Reply:
will propagate this change to other types where possible